Yosemite/Yosemite/Model/Storage/WooShippingCarrierPredefinedOptions+ReadOnlyConvertible.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+import Storage
+
+extension Storage.WooShippingCarrierPredefinedOptions {
+    var predefinedOptionsArray: [Storage.WooShippingPredefinedOption] {
+        return predefinedOptions?.toArray() ?? []
+    }
+}
+
+// Storage.WooShippingCarrierPredefinedOptions: ReadOnlyConvertible Conformance.
+//
+extension Storage.WooShippingCarrierPredefinedOptions: ReadOnlyConvertible {
+    /// Updates the Storage.WooShippingCarrierPredefinedOptions with the a ReadOnly WooShippingCarrierPredefinedOptions.
+    ///
+    public func update(with carrierPredefinedOptions: Yosemite.WooShippingCarrierPredefinedOptions) {
+        self.carrierID = carrierPredefinedOptions.carrierID
```

Comment:
Should we also set `predefinedOptions` ?

Reply:
@bozidarsevo In the `update` methods, we only update the Core Data entity's attributes. In this case, for example, the `Storage.WooShippingCarrierPredefinedOptions` entity only has the attribute `carrierID`.

Other properties in the read-only model that are relationships in the Core Data model are stored and updated separately, when we upsert the entities in `WooShippingStore`. We insert those child storage entities, update them, and save the relationship with the parent entity. You can see how that's handled for `WooShippingCarrierPredefinedOptions` and its `WooShippingPredefinedOption`s [in the store here](https://github.com/woocommerce/woocommerce-ios/blob/522db1b99a222ec85f9eaa8217f0c0c78c67f148/Yosemite/Yosemite/Stores/WooShippingStore.swift#L260-L279).